fluent-bundle/src/resolver.ts

```diff
@@ -198,13 +238,25 @@ function TermReference(scope, {name, attr, args}) {
 }
 
 // Resolve a call to a Function with positional and key-value arguments.
-function FunctionReference(scope, {name, args}) {
+function FunctionReference(
+  scope: Scope,
+  { name, args }: RuntimeFunctionReference
+): FluentType {
   // Some functions are built-in. Others may be provided by the runtime via
   // the `FluentBundle` constructor.
-  const func = scope.bundle._functions[name] || builtins[name];
+  let func = scope.bundle._functions[name];
   if (!func) {
-    scope.reportError(new ReferenceError(`Unknown function: ${name}()`));
-    return new FluentNone(`${name}()`);
+    switch (name) {
```

Comment:
I think I liked the old way better, but I guess the way TS does exports it's not as easy to achieve?

Reply:
Yeah, I was getting an error about `typeof import(...)` not having an index signature. A workaround which I found was to use `let func = scope.bundle._functions[name] || (<any>builtins)[name];`, which is shorter and much closer to the old way. Do you think it's a better choice here than the explicit switch?

Reply:
... or add `NUMBER` and `DATE` to `_functions` in the bundle constructor?

Reply:
> 
> 
> Yeah, I was getting an error about `typeof import(...)` not having an index signature. A workaround which I found was to use `let func = scope.bundle._functions[name] || (<any>builtins)[name];`, which is shorter and much closer to the old way. Do you think it's a better choice here than the explicit switch?

Yeah no I guess I agree that the switch is better than that, because we don't lose type info that way. Another option which imo beats switch would be sth like:
```typescript
let func = scope.bundle._functions[name] || (
    name == "DATETIME" || name == "NUMBER" ? builtins[name] : null
);
```
There should also be a more generic version to achieve the type refinement necessary to use `name` as an index, but I'm not coming up with it rn 🤔

Reply:
I like @Pike's suggestion to deal with this when the bundle is constructed rather than later on, when translations are resolved. I implemented it in https://github.com/projectfluent/fluent.js/pull/436/commits/4d48ced773d600d0ecbcf51961539ef254b88231.